### PR TITLE
changed the set of e2e tests of source and pullsubscription e2e from v1beta1 to v1, added smoke test for each supported version of pullsubscription and source

### PR DESCRIPTION
**Pull request by capri-xiyue**
Helps with #1409 #1410 #1411 #1412

## Proposed Changes

- 🧽 changed the set of e2e tests of source and pullsubscription from v1beta1 to v1
- added smoke test for each supported version of pullsubscription and source
-

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -156,11 +156,25 @@ func TestChannelTracing(t *testing.T) {
...
+
+// TestSmokePullSubscriptionV1beta1 test we can create a v1beta1 PullSubscription to ready state
+func TestSmokePullSubscriptionV1beta1(t *testing.T) {
+	cancel := logstream.Start(t)
+	defer cancel()
+	SmokePullSubscriptionTestHelper(t, authConfig, "v1beta1")
+}
+
+// TestSmokePullSubscriptionV1alpha1 test we can create a v1alpha1 PullSubscription to ready state
+func TestSmokePullSubscriptionV1alpha1(t *testing.T) {
```
Comment (capri-xiyue) [inline test/e2e/e2e_test.go:174]: I will add some comment to indicate for each supported version of resources, we only write smoke tests

File: test/e2e/test_auditlogs.go [hunk 2]
```diff
@@ -33,8 +33,60 @@ import (
...
+
+	project := os.Getenv(lib.ProwProjectKey)
+
+	auditlogsName := helpers.AppendRandomString("auditlogs-e2e-test")
+	svcName := helpers.AppendRandomString(auditlogsName + "-event-display")
+	topicName := helpers.AppendRandomString(auditlogsName + "-topic")
+	resourceName := fmt.Sprintf("projects/%s/topics/%s", project, topicName)
+
+	if cloudAuditLogsSourceVersion == "v1alpha1" {
+		lib.MakeAuditLogsV1alpha1OrDie(client, lib.AuditLogsConfig{
```
Comment (nachocano) [inline test/e2e/test_auditlogs.go:50]: can you just create this lib.AuditLogsConfig once outside the if, and then use it in all the cases?

File: test/e2e/test_build.go [hunk 3]
```diff
@@ -35,8 +35,38 @@ import (
...
+	lib.MakeTopicWithNameIfItDoesNotExist(t, events.CloudBuildTopic)
+
+	buildName := helpers.AppendRandomString("build")
+	svcName := "event-display"
+
+	client := lib.Setup(t, true, authConfig.WorkloadIdentity)
+	defer lib.TearDown(client)
+
+	if cloudBuildSourceVersion == "v1alpha1" {
+		lib.MakeBuildV1alpha1OrDie(client, lib.BuildConfig{
```
Comment (nachocano) [inline test/e2e/test_build.go:50]: same with this one

File: test/e2e/test_pubsub.go [hunk 4]
```diff
@@ -36,8 +36,49 @@ import (
...
+	defer deleteTopic()
+
+	psName := topic + "-pubsub"
+	svcName := "event-display"
+
+	client := lib.Setup(t, true, authConfig.WorkloadIdentity)
+	defer lib.TearDown(client)
+
+	if cloudPubSubSourceVersion == "v1alpha1" {
+		lib.MakePubSubV1alpha1OrDie(client, lib.PubSubConfig{
```
Comment (nachocano) [inline test/e2e/test_pubsub.go:52]: same here

File: test/e2e/test_pullsubscription.go [hunk 5]
```diff
@@ -47,20 +43,33 @@ func SmokePullSubscriptionTestImpl(t *testing.T, authConfig lib.AuthConfig) {
...
-					ServiceAccountName: authConfig.ServiceAccountName,
-				},
-			},
-		}),
-		reconcilertestingv1beta1.WithPullSubscriptionSink(lib.ServiceGVK, svcName))
-	client.CreatePullSubscriptionOrFail(pullsubscription)
-
-	client.Core.WaitForResourceReadyOrFail(psName, lib.PullSubscriptionTypeMeta)
+	if pullsubscriptionVersion == "v1alpha1" {
+		lib.MakePullSubscriptionV1alpha1OrDie(client, lib.PullSubscriptionConfig{
```
Comment (nachocano) [inline test/e2e/test_pullsubscription.go:47]: here as well

File: test/e2e/test_pubsub.go [hunk 6]
```diff
@@ -36,8 +36,49 @@ import (
...
+		lib.MakePubSubV1beta1OrDie(client, lib.PubSubConfig{
+			SinkGVK:            metav1.GroupVersionKind{Version: "v1", Kind: "Service"},
+			PubSubName:         psName,
+			SinkName:           svcName,
+			TopicName:          topic,
+			ServiceAccountName: authConfig.ServiceAccountName,
+		})
+	} else if cloudPubSubSourceVersion == "v1" {
+		lib.MakePubSubOrDie(client, lib.PubSubConfig{
+			SinkGVK:            metav1.GroupVersionKind{Version: "v1", Kind: "Service"},
```
Comment (nachocano) [inline test/e2e/test_pubsub.go:69]: I saw a lib.ServiceGVK in other files, can you use that?

File: test/e2e/test_storage.go [hunk 7]
```diff
@@ -36,8 +36,51 @@ import (
...
+	ctx := context.Background()
+	project := os.Getenv(lib.ProwProjectKey)
+
+	bucketName := lib.MakeBucket(ctx, t, project)
+	defer lib.DeleteBucket(ctx, t, bucketName)
+	storageName := helpers.AppendRandomString(bucketName + "-storage")
+	svcName := helpers.AppendRandomString(bucketName + "-event-display")
+
+	if cloudStorageSourceVersion == "v1alpha1" {
+		lib.MakeStorageV1alpha1OrDie(client, lib.StorageConfig{
```
Comment (nachocano) [inline test/e2e/test_storage.go:54]: and this one

File: test/e2e/e2e_test.go (hunk 1 above)
Comment (nachocano) [reply test/e2e/e2e_test.go:174]: if you can add that, it'd be great

**Comment by nachocano** [comment]
@capri-xiyue left a few nits, and it should be ready to go

**Comment by nachocano** [comment]
/lgtm
/approve

Thanks @capri-xiyue !!!

**Pull request merged**